examples/pmp_convor.py

```diff
@@ -0,0 +1,212 @@
+
+fg = graph.FactorGraph(variables=dict(S=S, W=W, SW=SW, X=X))
+
+# %%
+variable_names_for_OR_factors = {}
+
+# Add ANDFactors
+for idx_img in tqdm(range(n_images)):
+    for idx_chan in range(n_chan):
+        for idx_s_height in range(s_height):
+            for idx_s_width in range(s_width):
+                for idx_feat in range(n_feat):
+                    for idx_feat_height in range(feat_height):
+                        for idx_feat_width in range(feat_width):
+                            idx_img_height = idx_feat_height + idx_s_height
+                            idx_img_width = idx_feat_width + idx_s_width
+                            SW_var = (
+                                "SW",
+                                idx_img,
+                                idx_chan,
+                                idx_img_height,
+                                idx_img_width,
+                                idx_feat,
+                                idx_feat_height,
+                                idx_feat_width,
+                            )
+
+                            variable_names = [
+                                ("S", idx_img, idx_feat, idx_s_height, idx_s_width),
+                                (
+                                    "W",
+                                    idx_chan,
+                                    idx_feat,
+                                    idx_feat_height,
+                                    idx_feat_width,
+                                ),
+                                SW_var,
+                            ]
+
+                            fg.add_factor_by_type(
+                                variable_names=variable_names,
+                                factor_type=logical.ANDFactor,
+                            )
+
+                            X_var = (idx_img, idx_chan, idx_img_height, idx_img_width)
+                            if X_var not in variable_names_for_OR_factors:
+                                variable_names_for_OR_factors[X_var] = [SW_var]
+                            else:
+                                variable_names_for_OR_factors[X_var].append(SW_var)
+
+
+# Add ORFactors
+for X_var, variable_names_for_OR_factor in variable_names_for_OR_factors.items():
+    fg.add_factor_by_type(
+        variable_names=variable_names_for_OR_factor + [("X",) + X_var],  # type: ignore
+        factor_type=logical.ORFactor,
+    )
+
+for factor_type, factors in fg.factors.items():
+    print(f"The factor graph contains {len(factors)} {factor_type}")
+
+# %% [markdown]
+# ### Run inference and visualize results
+
+# %% [markdown]
+# PMP perturbs the model by adding Gumbel noise to unary potentials, then samples from the joint posterior *p(S,W | X)*.
+
+# %%
+run_bp, get_bp_state, get_beliefs = graph.BP(fg.bp_state, 3000)
+
+# %% [markdown]
+# We first compute the evidence without perturbation, similarly to the PMP paper
+
+# %%
+from scipy.special import logit as invsigmoid
```

Comment:
Move import to top of the notebook. Also can we just use `logit` instead of renaming it as `invsigmoid`? I did a quick search and don't see any other places doing this renaming.